lib/gui/models/store.js

```diff
@@ -287,10 +287,24 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
         throw errors.createError('Missing image size');
       }
 
-      if (!_.isNumber(action.data.size)) {
+      if (!_.isPlainObject(action.data.size)) {
         throw errors.createError(`Invalid image size: ${action.data.size}`);
       }
 
+      const MINIMUM_DRIVE_SIZE = 0;
+
+      if (!_.isInteger(action.data.size.original) || action.data.size.original < MINIMUM_DRIVE_SIZE) {
+        throw errors.createError(`Invalid original image size: ${action.data.size.original}`);
+      }
+
+      if (!_.isInteger(action.data.size.final.value) || action.data.size.final.value < MINIMUM_DRIVE_SIZE) {
```

Comment:
Probably not worth it for just two if-tests, but I guess if there were more you could abstract this into an `isPositiveInteger` function.